scripts/nix-profile-daemon.sh.in

```diff
@@ -0,0 +1,54 @@
+# Only execute this file once per shell.
+if [ -n "$__ETC_PROFILE_NIX_SOURCED" ]; then return; fi
+__ETC_PROFILE_NIX_SOURCED=1
```

Comment:
I don't think this is necessary. None of the below actions have too many side effects. This could make things confusing to someone who has for instance modified their NIX_PATH value and wants to "reset" it. They have to unset __ETC_PROFILE_NIX_SOURCED for that to work.

Reply:
There are lots of side effects below, although most of them do look to me like they are written to be idempotent. 

Setting the `PATH` is **not** idempotent, however, as it prepends to `${PATH}`, so running this multiple times would result in a `PATH` with multiple copies of the directories added by this script.